File: backend/src/main/java/com/campus/eventplatform/entity/Event.java
package com.campus.eventplatform.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("event")
public class Event {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String description;
    private String status; // DRAFT, PUBLISHED, CANCELLED, ENDED
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
    private Long creatorId;
    private Integer capacity;
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/event.ts
import api from '@/lib/api';

export interface Event {
  id: number;
  title: string;
  description: string;
  status: string;
  startTime: string;
  endTime: string;
  createTime: string;
  updateTime: string;
  creatorId: number;
  capacity: number;
  location?: string;
}

export const eventApi = {
  getEvents: () => api.get<{ code: number; data: Event[]; msg: string }>('/event/list'),
  getEventById: (id: number) => api.get<{ code: number; data: Event; msg: string }>(`/event/${id}`),
  createEvent: (data: Partial<Event>) => api.post<{ code: number; data: Event; msg: string }>('/event/create', data),
  updateEvent: (id: number, data: Partial<Event>) => api.put<{ code: number; data: Event; msg: string }>(`/event/${id}`, data),
  deleteEvent: (id: number) => api.delete<{ code: number; msg: string }>(`/event/${id}`),
};
